components/membership/registration-impl/src/main/kotlin/net/corda/membership/impl/registration/dynamic/member/DynamicMemberRegistrationService.kt

```diff
@@ -0,0 +1,432 @@
+    @Reference(service = LifecycleCoordinatorFactory::class)
+    private val coordinatorFactory: LifecycleCoordinatorFactory,
+    @Reference(service = CryptoOpsClient::class)
+    private val cryptoOpsClient: CryptoOpsClient,
+    @Reference(service = KeyEncodingService::class)
+    private val keyEncodingService: KeyEncodingService,
+    @Reference(service = CordaAvroSerializationFactory::class)
+    cordaAvroSerializationFactory: CordaAvroSerializationFactory,
+    @Reference(service = MembershipGroupReaderProvider::class)
+    private val membershipGroupReaderProvider: MembershipGroupReaderProvider,
+) : MemberRegistrationService {
+    /**
+     * Private interface used for implementation swapping in response to lifecycle events.
+     */
+    private interface InnerRegistrationService : AutoCloseable {
+        fun register(member: HoldingIdentity, context: Map<String, String>): MembershipRequestRegistrationResult
+    }
+
+    private companion object {
+        val logger: Logger = contextLogger()
+
+        const val PUBLICATION_TIMEOUT_SECONDS = 30L
+        const val SESSION_KEY_ID = "${PARTY_SESSION_KEY}.id"
+        const val SESSION_KEY_SIGNATURE_SPEC = "${PARTY_SESSION_KEY}.signature.spec"
+        const val LEDGER_KEY_ID = "${LEDGER_KEYS_KEY}.id"
+        const val LEDGER_KEY_SIGNATURE_SPEC = "${LEDGER_KEYS_KEY}.signature.spec"
+        const val REGISTRATION_ID = "corda.registration.id"
+        const val MEMBERSHIP_P2P_SUBSYSTEM = "membership"
+        const val PLATFORM_VERSION_CONST = "5000"
+        const val SOFTWARE_VERSION_CONST = "5.0.0"
+        const val SERIAL_CONST = "1"
+
+        val defaultCodeNameToSpec = mapOf(
+            ECDSA_SECP256K1_CODE_NAME to SignatureSpec("SHA512withECDSA"),
+            ECDSA_SECP256R1_CODE_NAME to SignatureSpec("SHA512withECDSA"),
+            EDDSA_ED25519_TEMPLATE to SignatureSpec.EDDSA_ED25519,
+            GOST3410_GOST3411_TEMPLATE to SignatureSpec.GOST3410_GOST3411,
+            RSA_CODE_NAME to SignatureSpec.RSA_SHA512,
+            SM2_CODE_NAME to SignatureSpec.SM2_SM3,
+            SPHINCS256_CODE_NAME to SignatureSpec.SPHINCS256_SHA512,
+        )
+    }
+
+    // for watching the config changes
+    private var configHandle: AutoCloseable? = null
+
+    // for checking the components' health
+    private var componentHandle: RegistrationHandle? = null
+
+    private var _publisher: Publisher? = null
+
+    /**
+     * Publisher for Kafka messaging. Recreated after every [MESSAGING_CONFIG] change.
+     */
+    private val publisher: Publisher
+        get() = _publisher ?: throw IllegalArgumentException("Publisher is not initialized.")
+
+    // Component lifecycle coordinator
+    private val coordinator = coordinatorFactory.createCoordinator(lifecycleCoordinatorName, ::handleEvent)
+
+    private val registrationRequestSerializer: CordaAvroSerializer<MembershipRegistrationRequest> =
+        cordaAvroSerializationFactory.createAvroSerializer { }
```

Comment:
Can you update the two usages of `createAvroSerializer` to log some warning? For example:
```suggestion
        cordaAvroSerializationFactory.createAvroSerializer { logger.warn("foo bar") }
```